src/xma/src/xmaapi/xma_utils.cpp

```diff
@@ -610,51 +676,56 @@ int32_t check_all_execbo(XmaSession s_handle) {
                 int32_t val = *ebo_it;
                 auto& ebo = priv1->kernel_execbos[val];
                 if (ebo.in_use) {
-                    ert_start_kernel_cmd *cu_cmd = 
-                        (ert_start_kernel_cmd*)ebo.data;
-                    if (cu_cmd->state == ERT_CMD_STATE_COMPLETED) {
+                    ert_start_kernel_cmd *cu_cmd_pkt = 
+                        reinterpret_cast<ert_start_kernel_cmd*>(ebo.xrt_run.get_ert_packet());// only for PS Kernels, will be removed after PS kernel migration to xrt::kernel 
```

Comment:
I am unsure you need a `reinterpret_cast` here
```suggestion
                    // only for PS Kernels, will be removed after PS kernel migration to xrt::kernel 
                    auto cu_cmd_pkt = static_cast<ert_start_kernel_cmd*>(ebo.xrt_run.get_ert_packet());
```

Reply:
seeing compilation error if we use static_cast here.  using auto now, not repeating the type. submitted latest changes in the PR.